src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AsyncStoragePoolConfig.java

```diff
@@ -0,0 +1,40 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import org.elasticsearch.common.settings.Settings;
+
+public class AsyncStoragePoolConfig {
```

Comment:
Will this class be used outside of `AsyncStoragePool`? The current usage is limited to `AsyncStoragePool`, why not to make it `static` inside `AsyncStoragePool`?

Reply:
Yes. It will be used outside AsyncStoragePool to notify config changes. Moreover having it static inner class will make an outer class dependent on inner class. Eg: AsyncStoragePool constructor accepting inner class AsyncStoragePoolConfig dependency.

Reply:
Will it only notify `AsyncStoragePool` about config changes? The class name `AsyncStoragePoolConfig ` suggests that it is specific to `AsyncStoragePool` and can't be used with other thread pools. In this case, I don't see anything wrong with making it more explicit and making `AsyncStoragePool` constructor accept an instance of an inner **static** class. It is simply a way to organize naming.

Reply:
I will rename it to ThreadPoolConfig to make it more generic if we want to use config for other thread pools.